### PR TITLE
Jd/add params to results

### PR DESCRIPTION
**Pull request by jd-lara**

### CODE REVIEW

File: src/core/operations_problem.jl [hunk 1]
```diff
@@ -470,6 +468,41 @@ function _build!(
...
+function get_dual_values(op_m::OperationsProblem; kwargs...)
+    cons = get(kwargs, :constraints_duals, nothing)
+    isnothing(cons) && return Dict{Symbol, Any}()
+    results_dict = Dict{Symbol, DataFrames.DataFrame}()
+    for c in cons
+        v = get_constraint(op_m.psi_container, c)
+        results_dict[c] = axis_array_to_dataframe(v)
+    end
+    return results_dict
+end
```
Comment (daniel-thom) [inline src/core/operations_problem.jl]: This returns different types. Maybe the following is more readable?

```
    cons = get(kwargs, :constraints_duals, [])
    return Dict(c => axis_array_to_dataframe(get_constraint(op_m.psi_container, c)) for c in cons)
```

File: src/core/operations_problem.jl [hunk 2]
```diff
@@ -470,6 +468,41 @@ function _build!(
...
+
+function get_parameters_value(op_m::OperationsProblem)
+    # TODO: Still not obvious implementation since it needs to get the multipliers from
+    # the system
+    params_dict = Dict{Symbol, DataFrames.DataFrame}()
+    parameters = get_parameters(op_m.psi_container)
+    isnothing(parameters) && return params_dict
+    isempty(parameters) && return params_dict
```
Comment (daniel-thom) [inline src/core/operations_problem.jl]:
```suggestion
    isnothing(parameters) || isempty(parameters) && return params_dict
```

File: src/core/psi_container.jl [hunk 3]
```diff
@@ -290,6 +288,10 @@ function _encode_for_jump(name::Symbol)
     return name
 end
 
+function decode_symbol(name::Symbol)
```
Comment (daniel-thom) [inline src/core/psi_container.jl]: The encode and decode functions should be symmetric.  We should change `_encode_for_jump` to `encode_symbol`.  And maybe `_JUMP_NAME_DELIMITER` needs a better name.

File: src/core/operations_problem.jl [hunk 4]
```diff
@@ -470,6 +468,41 @@ function _build!(
...
+    for (k, v) in parameters
+        !isa(v.update_ref, UpdateRef{<:PSY.Component}) && continue
+        params_key_tuple = decode_symbol(k)
+        params_dict_key = Symbol(params_key_tuple[1], "_", params_key_tuple[3])
```
Comment (daniel-thom) [inline src/core/operations_problem.jl]: Shouldn't this use the existing delimiter const?

File: src/core/simulation_results.jl [hunk 5]
```diff
@@ -215,8 +303,8 @@ function load_simulation_results(
     obj_value = Dict{Symbol, Any}(:OBJECTIVE_FUNCTION => optimizer["obj_value"])
     if !isempty(dual)
         duals = _read_references(duals, dual, stage, step, references, time_length)
-        results =
-            DualResults(variables, obj_value, optimizer, time_stamp, duals, results_folder)
+        results = 0.0
```
Comment (daniel-thom) [inline src/core/simulation_results.jl]: Can you insert a comment on why this is disabled?
Ditto, below.

File: src/core/operations_problem.jl (hunk 1 above)
Comment (jd-lara) [reply src/core/operations_problem.jl]: @Lilyhanig consider this in your PR to this branch, please.

File: src/core/operations_problem.jl (hunk 4 above)
Comment (jd-lara) [reply src/core/operations_problem.jl]: not in this case since it it for the printing of results

File: src/core/psi_container.jl [hunk 6]
```diff
@@ -499,14 +500,23 @@ function iterate_parameter_containers(psi_container::PSIContainer)
...
+        !isa(v.update_ref, UpdateRef{<:PSY.Component}) && continue
+        params_key_tuple = decode_symbol(k)
+        params_dict_key = Symbol(params_key_tuple[1], "_", params_key_tuple[3])
+        params_dict[params_dict_key] = axis_array_to_dataframe(get_parameter_array(v))
     end
-    return results_dict
+    return params_dict
+end
+
+function is_milp(container::PSIContainer)
```
Comment (claytonpbarrows) [inline src/core/psi_container.jl]: It looks like this is used to check if there are duals available. Should we generalize to any non LP?

Comment (jd-lara) [reply src/core/psi_container.jl]: There is still value in the duals of constraints of NLP's. The issue is that this broke a lot of things in the export of results before we were exporting .feather files with a bunch of invalid values.

File: src/core/simulation_results.jl [hunk 7]
```diff
@@ -241,13 +241,13 @@ for the desired step range and variables
 - `SimulationResultsReference::SimulationResultsReference`: the container for the reference dictionary created in execute!
 - `stage_number::Int = 1``: The stage of the results getting parsed: 1 or 2
 - `step::Array{String} = ["step-1", "step-2", "step-3"]`: the steps of the results getting parsed
-- `variable::Array{Symbol} = [:P_ThermalStandard, :P_RenewableDispatch]`: the variables to be parsed
+- `variable::Array{Symbol} = [:P__ThermalStandard, :P_RenewableDispatch]`: the variables to be parsed
```
Comment (daniel-thom) [inline src/core/simulation_results.jl]: Does this need to be `P__RenewableDispatch`?

File: test/test_operations_solve.jl [hunk 8]
```diff
@@ -233,12 +233,12 @@ res = solve_op_problem!(op_problem; constraints_duals = duals)
     end
 end
 
-path = joinpath(pwd(), "test_writing")
+path = joinpath(pwd(), "Test_writing")
```
Comment (daniel-thom) [inline test/test_operations_solve.jl]: Was this change intentional?  It's kinda weird to mix capital letters and underscores.

File: src/core/operations_problem.jl [hunk 9]
```diff
@@ -470,6 +468,30 @@ function _build!(
...
+    return results_dict
+end
+
+function get_dual_values(op_m::OperationsProblem; kwargs...)
+    cons = get(kwargs, :constraints_duals, [])
+    results_dict = Dict{Symbol, DataFrames.DataFrame}()
+    for c in cons
+        v = get_constraint(op_m.psi_container, c)
+        results_dict[c] = axis_array_to_dataframe(v)
+    end
```
Comment (daniel-thom) [inline src/core/operations_problem.jl]: Did you forget to delete this?

File: src/core/psi_container.jl [hunk 10]
```diff
@@ -537,3 +547,33 @@ function _write_psi_container(psi_container::PSIContainer, save_path::String)
...
 end
+
+function get_model_duals(op::PSIContainer, cons::Vector{Symbol})
+    results_dict = Dict{Symbol, DataFrames.DataFrame}()
+    for c in cons
+        v = get_constraint(op, c)
+        results_dict[c] = axis_array_to_dataframe(v)
+    end
+    return results_dict
+end
```
Comment (daniel-thom) [inline src/core/psi_container.jl]: This is redundant with `get_dual_values`.
